Frontend/src/app/graph-panel/page.tsx

```diff
@@ -0,0 +1,152 @@
+"use client";
+import React, { useState } from "react";
+import { InboxOutlined } from "@ant-design/icons";
+import type { UploadProps } from "antd";
+import {
+  Button,
+  Col,
+  Divider,
+  message,
+  Row,
+  Typography,
+  Upload,
+  Modal,
+  Input,
+} from "antd";
+import kafkaLOGO from "@/assets/images/kafka-logo.jpg";
+import hadoopLOGO from "@/assets/images/hadoop-logo.jpg";
+import Image from "next/image";
+import KafkaUploadModal from "@/components/graph-panel/kafka-upload-modal";
+import HadoopUploadModal from "@/components/graph-panel/hadoop-upload-modal";
+import { RcFile } from "antd/es/upload/interface";
+import { toast } from "react-toastify";
+import axios from "axios";
+
+const { Dragger } = Upload;
+
+export default function GraphUpload() {
+  const [kafkaModalOpen, setKafkaModelOpen] = useState<boolean>(false);
+  const [hadoopModalOpen, setHadoopModelOpen] = useState<boolean>(false);
+  const [file, setFile] = useState<File>();
+  const [fileUrl, setFileUrl] = useState<string>();
+  const [modalOpen, setModalOpen] = useState(false);
+  const [graphName, setGraphName] = useState<string>("");
+
+  const handleFileUpload = (file: RcFile) => {
+    setModalOpen(true);
+    setFile(file);
+
+    if (fileUrl) {
+      URL.revokeObjectURL(fileUrl);
+    }
+
+    if (file) {
+      const url = URL.createObjectURL(file);
+      setFileUrl(url);
+    } else {
+      setFileUrl(undefined);
+    }
+    return false;
+  };
+
+  const handleUpload = async () => {
+    if (!file) {
+      toast.error("Please select a file to upload");
+      return;
+    }
+
+      const formData = new FormData();
+      formData.append('file', file);    
+      formData.append('graphName', graphName); // Append the file name
+
+      // Send the file and filename with Axios POST request
+      axios.post('/backend/graph/upload', formData, {
+        headers: {
+          'Content-Type': 'multipart/form-data'
+        }
+      })
+      .then(response => {
+        console.log('File uploaded successfully', response.data);
+      })
+      .catch(error => {
+        console.error('Error uploading file', error);
+      });
+
+      setModalOpen(false);
+  };
+
+  return (
+    <div className="graph-upload-panel">
+      <Typography.Title level={4} style={{ margin: "20px 0px" }}>
+        Upload Graph Data:
+      </Typography.Title>
+      <Dragger
+        multiple={false}
+        maxCount={1}
+        beforeUpload={(file: RcFile) => handleFileUpload(file)}
+      >
+        <p className="ant-upload-drag-icon">
+          <InboxOutlined />
+        </p>
+        <p className="ant-upload-text">
+          Click or drag file to this area to upload
+        </p>
+        <p className="ant-upload-hint">
+          Support for a single or bulk upload. Strictly prohibited from
```

Comment:
Shall we remove the second sentence. I do not see any general applicability here.

Reply:
removed